File: app/src/main/java/me/rohank05/beat/command/commands/SkipCommand.java
package me.rohank05.beat.command.commands;

import me.rohank05.beat.command.ICommand;
import me.rohank05.beat.lavaplayer.PlayerManager;

import net.dv8tion.jda.api.EmbedBuilder;
import net.dv8tion.jda.api.entities.MessageEmbed;
import net.dv8tion.jda.api.events.interaction.command.SlashCommandInteractionEvent;


public class SkipCommand implements ICommand {
    @Override
    public void run(SlashCommandInteractionEvent event) {
        if (!CheckMusicPermission.checkRemainPermission(event)) return;
        if (!CheckMusicPermission.checkPermission(event)) return;
        MessageEmbed embed = new EmbedBuilder().setDescription("**Song Skipped**").setColor(16760143).build();
        event.getInteraction().getHook().sendMessageEmbeds(embed).queue();
        PlayerManager.getINSTANCE().getMusicManager(event.getGuild()).scheduler.nextTrack();
    }

    @Override
    public String getName() {
        return "skip";
    }
}

File: src/main/kotlin/com/therohankumar/commands/SkipCommand.kt
package com.therohankumar.commands

import com.therohankumar.audio.MusicManagerRegistry
import com.therohankumar.commands.util.CommandUtil
import net.dv8tion.jda.api.events.interaction.command.SlashCommandInteractionEvent
import net.dv8tion.jda.api.interactions.commands.OptionType
import net.dv8tion.jda.api.interactions.commands.build.Commands
import net.dv8tion.jda.api.interactions.commands.build.SlashCommandData
import org.springframework.stereotype.Component

@Component
class SkipCommand(private val registry: MusicManagerRegistry) : SlashCommand {

    override val name = "skip"

    override fun build(): SlashCommandData = Commands.slash(name, "Skip one or more tracks")
        .addOption(OptionType.INTEGER, "amount", "Number of tracks to skip (default 1)", false)

    override fun execute(event: SlashCommandInteractionEvent) {
        val manager = CommandUtil.requireSameVoice(event, registry) ?: return
        val amount = (event.getOption("amount")?.asInt ?: 1).coerceAtLeast(1)

        // Drop extra tracks from queue before advancing
        repeat(amount - 1) { manager.scheduler.queue.poll() }
        manager.scheduler.nextTrack()

        event.replyEmbeds(
            CommandUtil.successEmbed("Skipped", if (amount > 1) "Skipped $amount tracks." else null)
        ).queue()
    }
}
